vm/kvm-unit-tests/runtest.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+. /usr/bin/rhts_environment.sh
+
+export ACCEL=kvm
+unset ARCH
+unset STANDALONE
+cpus=$(grep -c ^processor /proc/cpuinfo)
+
+# test can only run on hardware that supports virtualization 
+if egrep -q '(vmx|svm)' /proc/cpuinfo; then
+    echo "Hardware supports virtualization, proceeding" | tee -a $OUTPUTFILE
+else
+    echo "Skipping test, CPU doesn't support virtualization" | tee -a $OUTPUTFILE
+    rhts-report-result $TEST SKIP $OUTPUTFILE
+    exit
+fi
+
+# test should only run on a system with 1 or more cpus
+if [ "$cpus" > 1 ]; then
+    echo "You have sufficient CPU's to run the test" | tee -a $OUTPUTFILE
+else
+    echo "Skipping test, system requires > 1 CPU" | tee -a $OUTPUTFILE
+    exit
+fi
+
+# test is only supported on x86_64
+if [ "$(uname -i)" == "x86_64" ]; then
+    echo "Running on x86_64 arch" | tee -a $OUTPUTFILE
+else
+    echo "Skipping test, test is only supported on x86_64" | tee -a $OUTPUTFILE
+    rhts-report-result $TEST SKIP $OUTPUTFILE
+    exit
+fi
+
+# set the virt kernel parameters
+echo -e "options kvm force_emulation_prefix=1\noptions kvm enable_vmware_backdoor=1" > /etc/modprobe.d/kvm-ci.conf
+
+# reload the modules
+rmmod kvm_intel kvm_amd kvm
+modprobe kvm_intel kvm_amd kvm
+
+KVM_SYSFS=/sys/module/kvm/parameters/
+KVM_OPTIONS="enable_vmware_backdoor force_emulation_prefix"
+
+for opt in $KVM_OPTIONS; do
+  if [ ! -f "$KVM_SYSFS/$opt" ]; then
+    echo "kernel option $opt not set" | tee -a $OUTPUTFILE
+    exit 1
+  else
+    echo "kernel option $opt is set" | tee -a $OUTPUTFILE
+  fi
+done
+
+# set the qemu-kvm path
+if [ -e /usr/libexec/qemu-kvm ]; then
+	export QEMU="/usr/libexec/qemu-kvm"
+fi
+
+if [ -z "$QEMU" ]; then
+	echo "Can't find qemu binary" | tee -a $OUTPUTFILE
+	report_result "qemu_not_found" "FAIL" 1
+	exit 1
+fi
+
+
+
+# if running on rhel8, use python3
+if grep --quiet 8.0 /etc/redhat-release;then
+    ln -s /usr/libexec/platform-python /usr/bin/python
+fi
+
+# enable nmi_watchdog since some unit tests depend on this
+echo 0 > /proc/sys/kernel/nmi_watchdog
+
+# clone the upstream kvm-unit-tests
+git clone git://git.kernel.org/pub/scm/virt/kvm/kvm-unit-tests.git
```

Comment:
I'd prefer we have a check if this command completed successfully (and abort if not). It's an external service and we can't predict any downtime. What do you think?

I also have a question purely out of curiosity since I'm not familiar with the test - is the master branch (which seems to be used) stable? Can we encounter any issues with tests being updated?

Reply:
@veruu we are picking a specific commit, will update periodically, also added some lines to abort if failed to clone/checkout.